File: packages/python/daytona/process.py
from typing import Optional
from .client import ExecuteResponse, WorkspaceInstance, WorkspaceToolboxApi
from .workspace import WorkspaceCodeToolbox

class Process:
    def __init__(self, code_toolbox: WorkspaceCodeToolbox, toolbox_api: WorkspaceToolboxApi, 
                 instance: WorkspaceInstance):
        self.code_toolbox = code_toolbox
        self.toolbox_api = toolbox_api
        self.instance = instance
    
    async def process_execute_command(self, command: str) -> ExecuteResponse:
        return await self.toolbox_api.process_execute_command(
            workspace_id=self.instance.id,
            project_id='main',  # todo: remove this after project refactor
            params={
                'command': command
            }
        )
    
    async def code_run(self, code: str) -> ExecuteResponse:
        run_command = self.code_toolbox.get_run_command(code)
        
        return await self.toolbox_api.process_execute_command(
            workspace_id=self.instance.id,
            project_id='main',  # todo: remove this after project refactor
            params={
                'command': run_command
            }
        )

File: packages/ts/src/Process.ts
import {
  ExecuteResponse,
  Workspace as WorkspaceInstance,
  WorkspaceToolboxApi,
} from './client'
import { WorkspaceCodeToolbox } from './Workspace'

export class Process {
  constructor(
    private readonly codeToolbox: WorkspaceCodeToolbox,
    private readonly toolboxApi: WorkspaceToolboxApi,
    private readonly instance: WorkspaceInstance,
  ) {}

  public async processExecuteCommand(
    command: string,
  ): Promise<ExecuteResponse> {
    return this.toolboxApi.processExecuteCommand({
      workspaceId: this.instance.id,
      projectId: 'main', //  todo: remove this after project refactor
      params: {
        command,
      },
    })
  }

  public codeRun(code: string): Promise<ExecuteResponse> {
    const runCommand = this.codeToolbox.getRunCommand(code)

    return this.toolboxApi.processExecuteCommand({
      workspaceId: this.instance.id,
      projectId: 'main', //  todo: remove this after project refactor
      params: {
        command: runCommand,
      },
    })
  }
}
